src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -182,22 +184,28 @@ public CourseDetailsBundle getCourseDetails(String courseId) throws EntityDoesNo
     }
 
     public List<String> getSectionsNameForCourse(String courseId) throws EntityDoesNotExistException {
+        return getSectionsNameForCourse(courseId, true);
+    }
 
-        verifyCourseIsPresent(courseId);
-        
+    private List<String> getSectionsNameForCourse(
+        String courseId, boolean checkExists) throws EntityDoesNotExistException {
+        if (checkExists) {
+            verifyCourseIsPresent(courseId);    
```

Comment:
You can change the function parameter to take in CourseAttributes and assert it is not null.
There won't be a need to call verifyCourseIsPresent then.